### DISCUSSION TITLE
Why every distro is having its own kernel and initrd?

### DISCUSSION THREAD

**Discussion by johnjcool** [kairos-io/kairos#1838] [General]
Why every distro is having its own kernel and initrd? What is the difference? Isn't it possible to use the same for every distro. if it will be the same you can save some download time, instead of downloading this for every distro.

cool project ;-)

**Comment by Itxaka** [comment]
Potentially yes, it should be possible to select just one kernel and use that in all the distros, but usually there is patches associated with their own distros for their own quirks, so in reality it could get a bit messy if we were to just use one, even if it was the upstream kernel directly.


We already ship alpine with the opensuse and Ubuntu kernels instead of using its own due our old boot mechanism being reliant and systems, so it's possible to do so.

Regarding the united, we kind of need to rebuild it to ship our immutability layer on it and it needs to be synced to the kernel and modules so same answer.

But yeah, theoretically nothing prevents us to ship just one kernel as a package and include it in the framework for all distros 😃

**Reply by johnjcool** [reply]
sounds nice :-) hope to see in the future. 

for the moment i'am struggeling with the net boot. i have tried different combinations but nothing is working with my hardware :-( i tried core and standard v2.3.0, v2.3.1, v2.4.0, debian, suse leap, rockylinux, ubuntu, fedora). in most cases i run in install initrd problems with /run/overlayfs not found and timeouts. in some cases i end up in grub bootloader after install which tells me the kernel has to boot first. the problem exists with auroraboot and pixiecore. if install with usb stick it works... what i'am doing wrong? can you tell me a combination which is working for the moment? i also tried to set some kernel rd.live* parameters but nothing helped.
